core/src/main/java/hudson/scm/SCM.java

```diff
@@ -326,7 +328,34 @@ public SCMRevisionState calcRevisionsFromBuild(AbstractBuild<?,?> build, Launche
     public SCMRevisionState _calcRevisionsFromBuild(AbstractBuild<?, ?> build, Launcher launcher, TaskListener listener) throws IOException, InterruptedException {
         return calcRevisionsFromBuild(build, launcher, listener);
     }
-    
+
+
+    /**
+     * Compares the current state of the remote repository against the given baseline {@link SCMRevisionState}.
+     * <p>
+     * This method is intended to <i>estimate</i> if the remote repository state has changed compared to current
+     * baseline, as a lightweight check, and as a result can trigger a finer grained comparison using
+     * {@link #compareRemoteRevisionWith(Job, Launcher, FilePath, TaskListener, SCMRevisionState)} if the SCM
+     * implementation do claim it require a workspace - see ${@link #requiresWorkspaceForPolling()}.
+     * <p>
+     * If the lightweight check can determine there's no change, returning ${@link PollingResult.Change#NONE}, then
+     * the polling mechanism can just stop and won't need to acquire a workspace.
+     * <p>
+     * If the lightweight check can determine changes, returning ${@link PollingResult.Change#SIGNIFICANT}, and
+     * ${@link #requiresWorkspaceForPolling()} return <code>false</code>, a full polling cycle will be executed,
+     * invoking {@link #compareRemoteRevisionWith(Job, Launcher, FilePath, TaskListener, SCMRevisionState)}.
+     * with a workspace.
+     * @since TODO
+     */
+    public PollingResult compareRemoteRevisionWith(@Nonnull Job<?,?> project, @Nullable Launcher launcher, @Nonnull TaskListener listener, @Nonnull SCMRevisionState baseline) throws IOException, InterruptedException {
+        if (requiresWorkspaceForPolling())
+             return PollingResult.SIGNIFICANT;
+        else
+             // for backward compatibility, we invoke compareRemoteRevisionWith with workspace=null
+             return compareRemoteRevisionWith(project, launcher, null, listener, baseline);
```

Comment:
yes, the new implementation only calls this method if a workspace is allocated. So plugins to rely on this version don't need to check for null